universal-application-tool-0.0.1/app/services/cloud/azure/Credentials.java

```diff
@@ -0,0 +1,32 @@
+package services.cloud.azure;
+
+import com.azure.identity.ChainedTokenCredential;
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import javax.inject.Inject;
+import org.mockito.Mockito;
+import play.Environment;
+
+/**
+ * This class retrieves Azure credentials through default provider.
+ *
+ * <p>See
+ * https://docs.microsoft.com/en-us/java/api/overview/azure/identity-readme?view=azure-java-stable
+ * for more details.
+ */
+public class Credentials {
```

Comment:
question (blocking): is this meant to be a singleton?

Reply:
Yup. fixed in latest commit.